File: greatest_candies.rb
def kids_with_candies(candies, extra_candies)
    arr = Array.new(candies.size, false)
    candies.each_with_index do |c, idx|
        if (c + extra_candies) >= candies.max
            arr[idx] = true
        else
            arr[idx] = false
        end
    end
    arr
end

File: greatest_candies.js
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function (candies, extraCandies) {
  const res = [];
  candies.forEach((num) => {
    let det = num + extraCandies;
    det >= Math.max(...candies) ? res.push(true) : res.push(false);
  });
  return res;
};
